src/coreclr/tools/aot/ILCompiler.ReadyToRun/TypeSystem/Mutable/MutableModule.cs

```diff
@@ -211,7 +211,6 @@ public MetadataReader Reader
 
                         byte[] metadataArrayTemp = _currentBinaryEmitter.EmitToMetadataBlob();
                         byte[] metadataArray = GC.AllocateArray<byte>(metadataArrayTemp.Length, pinned: true);
-                        System.Runtime.InteropServices.GCHandle.Alloc(metadataArray, System.Runtime.InteropServices.GCHandleType.Pinned);
```

Comment:
This seems a bit odd given it is stored anywhere, but it is also going to be creating a side effect. Do we know if this is okay to remove?

Reply:
The array is pinned in a line above. This handle would be leaked permanently.

Reply:
Is it guaranteed that the array is kept alive for as long as the unmanaged pointer into the array is in use (the unmanaged pointer is acquired a few lines below)?

Reply:
Realized it's not that simple. The handle is also keeping the array alive.
I'd like to leave it as-is and address in the future. The usages of GCHandle and PEReader don't look consistent in R2R.

Reply:
Could it just be a normal handle instead though?

Reply:
That won't make any difference.